core/src/banking_stage/transaction_scheduler/scheduler_metrics.rs

```diff
@@ -332,3 +334,71 @@ impl SchedulerTimingMetricsInner {
         self.receive_completed_time_us = 0;
     }
 }
+
+#[derive(Default)]
+pub struct SchedulerLeaderDetectionMetrics {
+    inner: Option<SchedulerLeaderDetectionMetricsInner>,
+}
+
+pub struct SchedulerLeaderDetectionMetricsInner {
+    slot: Slot,
+    bank_detected_time: Instant,
+    bank_detected_delay_us: u64,
+}
+
+impl SchedulerLeaderDetectionMetrics {
+    pub fn update_and_maybe_report(&mut self, bank_start: Option<&BankStart>) {
+        match (&self.inner, bank_start) {
+            (None, Some(bank_start)) => self.initialize_inner(bank_start),
+            (Some(_inner), None) => self.report_and_reset(),
+            (Some(inner), Some(bank_start)) if inner.slot != bank_start.working_bank.slot() => {
+                self.report_and_reset();
+                self.initialize_inner(bank_start);
+            }
+            _ => {}
+        }
+    }
+
+    fn initialize_inner(&mut self, bank_start: &BankStart) {
+        let bank_detected_time = Instant::now();
+        let bank_detected_delay_us = bank_detected_time
+            .duration_since(*bank_start.bank_creation_time)
+            .as_micros()
+            .try_into()
+            .unwrap_or(u64::MAX);
```

Comment:
i prefer doing this calc in `report_and_reset` and holding `bank_start.bank_creation_time` as `self.bank_created_time`. in this way, all the calc are grouped in a single place and holding 2 Instants rather than 1 Instant and 1 Duration (us in u64 actually) is simpler imo.

Reply:
768ca1a88f28f36d59a90c27e346538260004d34